### PR TITLE
Fixed GPS Sensor and Compass usage for baselines using float32 for GPS & Compass sensors

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
Fixed GPS Sensor and Compass usage for baselines in https://github.com/facebookresearch/habitat-lab/issues/604. Specifically the PR is fixing error:
`RuntimeError: expected scalar type Float but found Double`

## How Has This Been Tested

Added additional tests to avoid issues in future.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: habitat_baselines/config/test/ddppo_pointnav_all_sensors_test.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+BASE_TASK_CONFIG_PATH: "configs/test/habitat_all_sensors_test.yaml"
+TRAINER_NAME: "ddppo"
+ENV_NAME: "NavRLEnv"
+SIMULATOR_GPU_ID: 0
+TORCH_GPU_ID: 0
+VIDEO_OPTION: []
+TENSORBOARD_DIR: ""
+EVAL_CKPT_PATH_DIR: "data/test_checkpoints/ddppo/pointnav/ckpt.0.pth"
```
Comment (erikwijmans) [inline habitat_baselines/config/test/ddppo_pointnav_all_sensors_test.yaml:8]:
```suggestion
EVAL_CKPT_PATH_DIR: "data/test_checkpoints/ddppo/pointnav-all/ckpt.0.pth"
```

File: habitat_baselines/config/test/ddppo_pointnav_all_sensors_test.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+NUM_ENVIRONMENTS: 1
+CHECKPOINT_FOLDER: "data/test_checkpoints/ddppo/pointnav/"
```
Comment (erikwijmans) [inline habitat_baselines/config/test/ddppo_pointnav_all_sensors_test.yaml:10]:
```suggestion
CHECKPOINT_FOLDER: "data/test_checkpoints/ddppo/pointnav-all/"
```

**Comment by erikwijmans** [comment]
CI is failing because checkpoints are getting overwritten

**Review by Skylion007 — APPROVED** [review]

**Comment by mathfac** [comment]
Closing in favor of https://github.com/facebookresearch/habitat-lab/pull/606 as this branch isn't expected to be rebased on latest master.

**Pull request closed without merging**